src/logion/services/extrinsicsbuilder.ts

```diff
@@ -46,17 +44,25 @@ export class ExtrinsicsBuilder {
                         method: event.method,
                         data: event.data,
                     };
-                    if(event.section === "logionLoc" && event.method === "StorageFeeWithdrawn") {
-                        extrinsicBuilder.storageFee = this.getFee(jsonEvent);
-                    }
-                    if(event.section === "logionLoc" && event.method === "LegalFeeWithdrawn") {
-                        extrinsicBuilder.legalFee = this.getLegalFee(jsonEvent);
-                    }
-                    if(event.section === "logionLoc" && event.method === "CertificateFeeWithdrawn") {
-                        extrinsicBuilder.certificateFee = this.getFee(jsonEvent);
-                    }
-                    if(event.section === "logionLoc" && event.method === "ValueFeeWithdrawn") {
-                        extrinsicBuilder.valueFee = this.getFee(jsonEvent);
+                    if(event.section === "logionLoc") {
+                        if(event.method === "StorageFeeWithdrawn") {
+                            extrinsicBuilder.storageFee = this.getFee(jsonEvent);
+                        }
+                        if(event.method === "LegalFeeWithdrawn") {
+                            extrinsicBuilder.legalFee = this.getLegalFee(jsonEvent);
+                        }
+                        if(event.method === "CertificateFeeWithdrawn") {
+                            extrinsicBuilder.certificateFee = this.getFee(jsonEvent);
+                        }
+                        if(event.method === "ValueFeeWithdrawn") {
+                            extrinsicBuilder.valueFee = this.getFee(jsonEvent);
+                        }
+                        if(event.method === "CollectionItemFeeWithdrawn") {
+                            extrinsicBuilder.collectionItemFee = this.getFee(jsonEvent);
+                        }
+                        if(event.method === "TokensRecordFeeWithdrawn") {
+                            extrinsicBuilder.tokensRecordFee = this.getFee(jsonEvent);
+                        }
```

Comment:
This method `getFee()` grabs the amount paid (typically by the requester), but ignores the amount received by the owner of the LOC, a percentage of this latter amount. Let's discuss the solution.

Here is the full definition of the 2 new events:
```rust
/// Issued when Collection Item Fee is withdrawn. [payerAccountId, fee, beneficiary, amountReceived]
CollectionItemFeeWithdrawn(T::AccountId, BalanceOf<T>, Beneficiary<T::AccountId>, BalanceOf<T>),
/// Issued when Token Record Fee is withdrawn. [payerAccountId, fee, beneficiary, amountReceived]
TokensRecordFeeWithdrawn(T::AccountId, BalanceOf<T>, Beneficiary<T::AccountId>, BalanceOf<T>),
```

Reply:
Indeed, done.